### PR TITLE
format()->String.Format

### PR DESCRIPTION
**Pull request by jewalky**
Also added AppendFormat. Because it's in FString, and also because "x += String.Format" is longer to type than "x.AppendFormat".
You may want to later add Printf or something like that to combine A_Log with String.Format.
That should be very straightforward, just look at the current implementation of the string methods.

### CODE REVIEW

File: wadsrc/static/zscript/shared/sharedmisc.txt [hunk 1]
```diff
@@ -159,7 +159,7 @@ class CustomSprite : Actor
 		String name;
 		Super.BeginPlay ();
 
-		format(name, "BTIL%04d", args[0] & 0xffff);
+		//format(name, "BTIL%04d", args[0] & 0xffff); // note: this does NOTHING and did NOTHING. what?
 		picnum = TexMan.CheckForTexture (name, TexMan.TYPE_Build);
```
Comment (Gaerzi) [inline wadsrc/static/zscript/shared/sharedmisc.txt]: Did you miss that line? name is used to find picnum.

Comment (jewalky) [inline wadsrc/static/zscript/shared/sharedmisc.txt]: I have no fucking idea what that line does.
The syntax for format is "format(<format string>, <format arguments>)".
Graf probably meant name = format("BTIL%04d", args[0]&0xffff), but I'm not sure about that and didn't even want to touch it — Graf can do it himself.

Comment (jewalky) [inline wadsrc/static/zscript/shared/sharedmisc.txt]: Side note: with UDMF, pretty sure args[0] can reach 5 digits even with clamping to WORD.

**Comment by coelckers** [comment]
Forget the CustomSprite. This was originally just a hack to show something for Build maps, which are broken anyway. Consider that thing deprecated, I don't think that anyone has ever used it in an actual project. As long as it works like it was originally intended it is ok.

**Pull request merged**